streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -590,6 +590,30 @@ public void prepareRecycle() {
         log.info("Closed and recycled state");
     }
 
+    @Override
+    public void preparePoll() {
+        if (state() == State.RUNNING) {
+            // before running a new poll on the main consumer, if a queue's buffered size has been
+            // decreased to the threshold, we can then resume the consumption on this partition
+            final Set<TopicPartition> currentAssignment = mainConsumer.assignment();
+
+            mainConsumer.resume(
```

Comment:
Why do we not do this in `DefaultTaskManager#unassignTask()`? At least for the new threading model. I guess this is because the consumer should not be accessed by the processing threads. Is this correct?

Alternatively, you could record the partitions that can be resumed in `DefaultTaskManager#unassignTask()` and then pass those partitions to the polling thread.

I just try to avoid to iterate at each poll over all tasks and partitions.

Reply:
Good point, I improved the implementation by adding some internal state to `StreamTask`, that stores the partitions that need to be resumed. That actually simplifies the code so it's probably the right thing to do.